src/helpers/calculateCarbon.ts

```diff
@@ -1,5 +1,33 @@
-import { CountryName } from "../constants/Countries";
+import { COUNTRIES, CountryName } from "../constants/Countries";
+
+export const calculateCarbon = (bytes: number, selectedCountries: Map<CountryName, number>): number => {
+    const _selectedCountries = new Map(selectedCountries);
+    let totalPercentage = 0;
+
+    let noOfCountriesMissingPercentages = 0;
+
+    _selectedCountries.forEach((value, key) => {
+        totalPercentage += value;
+        if (value === 0) {
+            noOfCountriesMissingPercentages += 1;
+        }
+    });
```

Comment:
If you're going to refactor the array `selectedCountries` param to be an array of objects, make sure to change the foreachs with reduces and remove the lets